### PR TITLE
Add native build support

### LINKED ISSUE #1198: Add Native Build Support
## Native Build Support
We should add support to the dotnet-monitor repo to build native code specifically to enable compiling test and produciton cor profilers in `dotnet-monitor`.

Ideally this repo should be able to build the test profilers provided in the runtime repo here: [dotnet/runtime: src/tests/profiler/native](https://github.com/dotnet/runtime/tree/main/src/tests/profiler/native)

We will also want to be able to build our own profiler to support features such as trigger on exception.

### PR DESCRIPTION
**Pull request by jander-msft**
This change adds native build support to the repository. It is broken down into three separate changes:
1) Copy (without changing) PAL, CMake support files, and native support scripts from dotnet/runtime @ 09f681f7aa481d441c45ebeb9807694aa6df547e
2) Copy build scripts from dotnet/diagnostics @ b6b2bb644102806864da991760b6d2a599c312d4
3) Make adjustments to files to successfully build PAL in repository.

This separation should make it easier to review such that you only really need to look at https://github.com/dotnet/dotnet-monitor/commit/649d5f74874295fbc193a93e703440b9bccc0b7f

closes #1198

### CODE REVIEW

File: eng/build.yml [hunk 1]
```diff
@@ -60,8 +61,9 @@ jobs:
     ${{ if eq(parameters.osGroup, 'Linux') }}:
       container: mcr.microsoft.com/dotnet-buildtools/prereqs:centos-7-20210714125435-9b5bbc2
 
+    # CMake + Clang is broken on Alpine 3.14 prereqs image
     ${{ if eq(parameters.osGroup, 'Linux_Musl') }}:
-      container: mcr.microsoft.com/dotnet-buildtools/prereqs:alpine-3.14-WithNode-20211214164112-c401c85
+      container: mcr.microsoft.com/dotnet-buildtools/prereqs:alpine-3.13-WithNode-20210910135845-c401c85
```
Comment (wiktork) [inline eng/build.yml:66]: Is this 3.14 to 3.13 change intentional?

Comment (wiktork) [reply eng/build.yml:66]: Nvm I see the comment.

File: eng/build.yml [hunk 2]
```diff
@@ -133,19 +130,29 @@ jobs:
...
         $(_InternalInstallArgs)
         $(_InternalBuildArgs)
-        $(_InternalSignArgs)
-        $(_InternalPublishArgs)
       displayName: Build and Test
       condition: succeeded()
-
-    # # Publish test results to Azure Pipelines
+    
+    # Native build does not support signing. Move pack, sign, and publish
```
Comment (wiktork) [inline eng/build.yml:142]: We still do authenticode on windows for native dll's right?

Comment (jander-msft) [reply eng/build.yml:142]: It should. We aren't building any dlls yet though, so I cannot verify that at this time.

**Pull request closed without merging**